packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   placeholder: undefined,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
+
+const dateState = { day: '', month: '', year: '' };
+
+const DatePickerInput = (props) => {
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    inputAttributes,
+    inputRefCallback,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const momentDateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormatOrder = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateSeparator = intl.formatMessage({ id: 'Terra.datePicker.separator' });
+  const label = ariaLabel || intl.formatMessage({ id: 'Terra.datePicker.date' });
+  const placeholderValues = DateUtil.getDateInputValues(dateFormatOrder, placeholder, dateSeparator);
+  const theme = React.useContext(ThemeContext);
+
+  const setDayRef = useCallback(node => {
+    dayInputRef.current = node;
+    if (inputRefCallback && dateFormatOrder === 'YMD') {
+      inputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, inputRefCallback]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  const setMonthRef = useCallback(node => {
+    monthInputRef.current = node;
```

Comment:
How com the Day / Year set ref callbacks check the dateFormatOrder, but the month does not? The default is month first, so shouldn't this check as well?

Reply:
The `inputRefCallback` was added to address the issue https://github.com/cerner/terra-framework/pull/1208#issuecomment-685150000.
So as per the expected behaviour the auto focus/movement should move from the last input of the date input to the time input. So according to this **only day and year** can be the last inputs for date input.
This is the reason why only day and year are passed onto inputRefCallback and also since this is being used only internally, i have made it a private prop.
Also, i guess ill add documentation for this logic as well to avoid future confusion.

Reply:
Updated documentation: https://github.com/cerner/terra-framework/pull/1208/commits/623265a07fb2323e62a269e3e72b183ee8c8b2da